moving-forward-series/simple-payment-verification.md

```diff
@@ -0,0 +1,50 @@
+
+## How does Decred’s SPV work (a simple example)
+
+Imagine we are a light wallet. Our job is to track user's funds, more specifically to track addresses and transactions (having those we can compute the balances). Where do we get the data from? The blocks. We need data from the blocks. But we have restrictions. Since we want to be light and fast, we would rather not download and store the entire blockchain. We also would like to be private and secure, so we can't ask centralised servers "give me data for this address" as it would help them map our stuff.
+
+So we require some smart way to get data from the blocks to meet those restrictions. Here's what Decred does:
+*  The first step is to identify which blocks have our data. We connect to full nodes directly and download these tiny smart pieces of data called "filters", one filter for each block. 
+* Why "filters", what are we "filtering"? We are filtering blocks.
+* For each filter, we take our wallet data and throw it at the filter to see if its associated full block has any data we're interested in. 
+* If the answer is no, we skip that block, never download or process it (“e.g. filter it out”). 
+* If the answer is yes, the block associated with the filter has something relevant for our wallet, with a very high probability. 
+* So we move to the second step — fetch that full block, find our data in it, possibly discover new addresses/transactions that we need to track, and update our records. 
+* Then we move to the next filter and repeat the process for every block in the blockchain.
+
+In essence, we "filter through" a very lightweight reflection of the full chain, and fetch our data as we discover where it is.
+
+## Compact Block Filters
+
+SPV relies on the ability to identify and retrieve relevant transactions from the blockchain. Filters act like a search mechanism to help an SPV wallet find transactions it should care about. Decred’s Compact filters allow SPV nodes to retrieve this information without having to reveal or leak information regarding coin ownership. For instance, SPV nodes can ask full node peers for blocks that contain matching data without revealing addresses, keys, or the transactions they are looking for. Decred's filtering process is completed in the wallet to enhance privacy. Older methods of filtering put the entire workload onto the full nodes running the network, which is bad for network efficiency and privacy.
```

Comment:
* The SPV model relies on the ability...
* Decred's compact filters allow SPV wallets to retrieve this information...
* For instance, SPV wallets can ask full node peers...


"SPV nodes" do not exist.